webapp/annotations/annotation.xml

```diff
@@ -0,0 +1,86 @@
+                            <PropertyValue Property="Label" String="{@i18n>CategoryText}" />
+                            <PropertyValue Property="Value" Path="Category/Name" />
+                        </Record>
+                        <Record Type="UI.DataField" >
+                            <PropertyValue Property="Label" String="{@i18n>SupplierText}" />
+                            <PropertyValue Property="Value" Path="Supplier/Name" />
+                        </Record>
+                    </Collection>
+                </Annotation>
+                <Annotation Term="UI.SelectionFields" >
+                    <Collection >
+                        <PropertyPath  >Price</PropertyPath>
+                        <PropertyPath  >Category/Name</PropertyPath>
+                        <PropertyPath  >Supplier/Name</PropertyPath>
+                    </Collection>
+                </Annotation>
+                
+            </Annotations>
+
+            <Annotations Target="ODataDemo.Category/Name">
+                <Annotation Term="Common.Label" String="{@i18n>CategoryText}" />
+                <Annotation Term="Common.ValueList">
+                    <Record Type="Common.ValueListType">
+                        <PropertyValue Property="CollectionPath" String="Categories" />
+                        <PropertyValue Property="Parameters">
+                            <Collection>
+                                <Record Type="Common.ValueListParameterInOut">
+                                    <PropertyValue Property="LocalDataProperty" PropertyPath="Name" />
+                                    <PropertyValue Property="ValueListProperty" String="Name" />
+                                </Record>
+                            </Collection>
+                        </PropertyValue>
+                    </Record>
+                </Annotation>
+                <Annotation Term="Common.ValueListWithFixedValues" />
+            </Annotations>
+            <Annotations Target="ODataDemo.Supplier/Name">
+                <Annotation Term="Common.Label" String="{@i18n>SupplierText}" />
+                <Annotation Term="Common.ValueList">
+                    <Record Type="Common.ValueListType">
+                        <PropertyValue Property="CollectionPath" String="Suppliers" />
+                        <PropertyValue Property="Parameters">
+                            <Collection>
+                                <Record Type="Common.ValueListParameterInOut">
+                                    <PropertyValue Property="LocalDataProperty" PropertyPath="Name" />
+                                    <PropertyValue Property="ValueListProperty" String="Name" />
+                                </Record>
+                            </Collection>
+                        </PropertyValue>
+                    </Record>
+                </Annotation>
+                <Annotation Term="Common.ValueListWithFixedValues" />
```

Comment:
Let's make supplier filter as value help dialog, not combobox.
The reason why it shouldn't be combobox is there could be plenty suppliers. Therefore combobox is not suitable for this case, but value help dialog with table inside it.

Reply:
Combobox is intended to be used when you're sure there couldn't be more than 100 records and data structure is simple.
Category filter is the best case to use combobox but not suppliers.